File: temp_edit_api_settings.py
from pathlib import Path
import sys

print('starting script')

try:
    path = Path('src/components/settings/ApiSettingsView.tsx')
    print('got path')
    data = path.read_text()
    print('read data')
    marker = "if (editorMode === null) {"
    start = data.index(marker)
    print('found marker at', start)
    ret = data.find("  return (\n    <SettingsViewShell>", start)
    print('found return at', ret)
    if ret == -1:
        raise RuntimeError('return block not found')
    official = "      {editorMode === 'official' ? ("
    official_start = data.find(official, ret)
    print('official start', official_start)
    if official_start == -1:
        raise RuntimeError('official block not found')
    third = "      {editorMode === 'third-party' ? ("
    third_start = data.find(third, official_start)
    print('third start', third_start)
    if third_start == -1:
        raise RuntimeError('third-party block not found')
    return_end = data.rfind("  );")
    print('return end', return_end)
    if return_end == -1:
        raise RuntimeError('return end not found')
    official_block = data[official_start:third_start]
    third_block = data[third_start:return_end]
    new_block = (
        "  return (\n"
        "    <SettingsViewShell>\n"
        f"{official_block}\n\n{third_block}\n"
        "    </SettingsViewShell>\n"
        "  );\n"
    )
    new_data = data[:ret] + new_block + data[return_end + len("  );") :]
    Path('temp_ApiSettingsView.tsx').write_text(new_data)
    print('wrote new file')
except Exception as exc:
    print('Error during script execution:', exc, file=sys.stderr)
    raise

File: temp_edit_api_settings.js
const fs = require('node:fs');
const path = 'src/components/settings/ApiSettingsView.tsx';
let data;
try {
  data = fs.readFileSync(path, 'utf-8');
} catch (err) {
  console.error('read error', err);
  process.exit(1);
}
const marker = 'if (editorMode === null) {';
const start = data.indexOf(marker);
if (start === -1) {
  throw new Error('marker not found');
}
const retIndex = data.indexOf('  return (\n    <SettingsViewShell>', start);
if (retIndex === -1) {
  throw new Error('return block not found');
}
const officialMarker = "      {editorMode === 'official' ? (";
const officialStart = data.indexOf(officialMarker, retIndex);
if (officialStart === -1) {
  throw new Error('official block not found');
}
const thirdMarker = "      {editorMode === 'third-party' ? (";
const thirdStart = data.indexOf(thirdMarker, officialStart);
if (thirdStart === -1) {
  throw new Error('third-party block not found');
}
const returnEnd = data.lastIndexOf('  );');
if (returnEnd === -1) {
  throw new Error('return end not found');
}
const officialBlock = data.slice(officialStart, thirdStart);
const thirdBlock = data.slice(thirdStart, returnEnd);
const newBlock = `  return (
    <SettingsViewShell>
${officialBlock}

${thirdBlock}
    </SettingsViewShell>
  );
`;
const before = data.slice(0, retIndex);
const after = data.slice(returnEnd + '  );'.length);
const newData = before + newBlock + after;
fs.writeFileSync('temp_ApiSettingsView.tsx', newData);
